File: aula 3/com.br.gft/src/com/objetos/Carro.java
package com.objetos;

public class Carro extends Veiculo {
	public Carro(String marca, String modelo, String placa, float km, boolean isLigado,
			int litrosCombustivel, int velocidade, double preco, int portas, int ano) {
		super(marca, modelo, placa, km, isLigado, litrosCombustivel, velocidade, preco);
		this.portas = portas;
		this.ano = ano;
	}
	private int portas;
	private int ano;
	
	
	public int getPortas() {
		return portas;
	}
	public void setPortas(int portas) {
		this.portas = portas;
	}
	public int getAno() {
		return ano;
	}
	public void setAno(int ano) {
		this.ano = ano;
	}
	
	@Override
	public void acelerar() {
		if (this.isLigado == true) {
			velocidade += 50;
			litrosCombustivel -= 1;
		} else if (this.litrosCombustivel < 1) {
			System.out.println("impossivel acelerar");
		} else {
			System.out.println("o veiculo está desligado");
		}
	}
	
	
	
	
}
	

File: C#/Veiculo/Veiculo/objetos/Carro.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Veiculo.objetos
{
    class Carro : Veiculo
    {
        private int portas;
        private int ano;

        public Carro(string marca, string modelo, string placa, float km, bool isLigado, int litrosCombustivel, int velocidade, double preco, int portas, int ano) : base(marca, modelo, placa, km, isLigado, litrosCombustivel, velocidade, preco)
        {
            this.Portas = portas;
            this.Ano = ano;
        }

        public int Portas { get => portas; set => portas = value; }
        public int Ano { get => ano; set => ano = value; }

        public void acelerar()
        {
            if (IsLigado == true)
            {
                Velocidade += 50;
                LitrosCombustivel -= 1;
            }
            else if (LitrosCombustivel < 1)
            {
                Console.WriteLine("impossivel acelerar");
            }
            else
            {
                Console.WriteLine("o veiculo está desligado");
            }
        }
    }
}
